jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/gcprules/APIKeys/RotateAPIKeysRule.java

```diff
@@ -96,7 +96,7 @@ private boolean rotateApiKeys(String vmEsURL, Map<String, Object> mustFilter) th
             logger.info("hit array size {}", hitsJsonArray.size());
             JsonObject apiKeys = (JsonObject) ((JsonObject) hitsJsonArray.get(0)).get(PacmanRuleConstants.SOURCE);
             String createdDateStr = apiKeys.get("createdDate").getAsString();
-            SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSSSSS");
+            SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:00Z");
```

Comment:
@plyubich this change here is not global, but for createdDate for gcp api keys, since this policy will evalutae only gcp asset type apikeys. This format change was done as part of PLG-476 at mapper side, we need to make it consistent here since otherwise the policy is failing.

Reply:
@shefalibisht1992 @AnandPaladin i thought the original mapper fix for GCP keys was supposed to already address the format, why is the mapper forcing a standard on an existing system?

Reply:
@plyubich the mapper change, makes gcp api keys in sync with the es data format that is used for other assets as well. The change includes- to make gcp apikeys follow the es date format and to add support in mapping for this format for createdDate so that it can be mapped a date attribute. The issue is since createdDate is an attribute common between gcp api keys and violations. All the violations entities have createdDate as date which enables sorting, filltering and other other action, but with createdDate of gcp apikeys as text, the violations raised against them do not support these. 
cc: @AnandPaladin

Reply:
@plyubich to add, the issue is happening because `createDate` column exists in the asset data. While loading the asset data, ES creates the datatype of that column as text. Once created, changing this is difficult we need to re-index to change the datatype/mapping for that column.

If any asset does not have the `createdDate` column, the violation would create the same with the default data format and ES is creating that as a date so that our filters and sort will not have any issues.

In case where asset data also contain `createdDate` as that is being mapped as text, violation filters and soritng will have issues. 

This change in shipper to make the `createdDate` to date in case that is exists in the asset data so that violations sorting and filters will not have the problem.

As long as in mapper we follow a specific date format, we should be good here. As the policy on API keys also uses a different format, we are trying to match it with what we changed in the mapper.

Let me know if you see any race conditions.

Reply:
@santhosh-challa @plyubich I believe we have solved this problem in [this PR](https://github.com/PaladinCloud/Plugins/pull/136). So `createdData` is supposed to be date, not string. Is it still a string for ES? `createdDate` now has the same format as `discoverydate` so it should work as a date, right?

Reply:
or shipper just doesn't know about createdDate being a date column?

Reply:
I see that somewhere it called `CreatedTime` and in other types (like API Key) it called `CreatedDate`, does it make sense to use the same naming? In GCP data it is `createdTime`. Will it solve this issue if we rename that column in API Keys?

Reply:
@mkinchin-paladin discoverydate is also string in ES. 
In ES, either we need to specify the attribute as date and provide the format we are going to use while creating index/assets. OR  the attribute value needs to be in a specific format(default format for date in ES) so that ES can map them as date. 

https://github.com/PaladinCloud/Plugins/pull/136 this pr did not update the format that to what ES expects (default format) for date, but rather made the format in sync with other asset data. Therefore  ES still maps the attribute as text, same for discoverydate.

Renaming the column could solve the issue since the root cause is the same attribute name 'createdDate' between violations and asset data. @santhosh-challa

Reply:
@mkinchin-paladin @plyubich Renaming the column would solve the problem, but we need to see if that is ok to change the source column from what we have the respective cloud provider. We also need to make sure we do this for any future asset types.

Reply:
@santhosh-challa ok, let me see if I can do that today. And then we will check if mapper change is all what we need for this task. My personal believe that if you have anything wrong with data then it's the mapper issue, not shipper. Shipper MUST NOT alter the data, it is done ONLY by mapper. Otherwise we will get lost where that piece of data was changed, in mapper, in shipper or anywhere else.